client/src/services/documentService.ts

```diff
@@ -0,0 +1,76 @@
+import axios from 'axios';
+
+declare var DATASTORE_URI: string;
+
+export const DocumentService = {
+  
+  getSampleDocument: (): Array<any> =>{
+    function getDocumentMd() : string {
+      var paragraph = "# Welcome to Wrattler. \
+This is a sample notebook showing no data science at all.\
+First, we create one frame in JavaScript:\n\
+```javascript\
+var one = [{'name':'Joe', 'age':50}]\
+```\
+Second, we create one frame in Python\
+```python\
+two = pd.DataFrame({'name':['Jane'], 'age':[52]})\
+```\
+Third we combine the two:\
+```r\
+three <- rbind(one,two) \
+```\
+";
+      return paragraph
+    }
+    let documents = []; 
+
+    function getCellCode(language: string, codeCell: string) {
+      let languageMarker = "```".concat(language)
+      let languageMarkerBegin = codeCell.indexOf(languageMarker)
+      if (languageMarkerBegin > -1)
+      {
+        let languageMarkerEnd = codeCell.indexOf("```", languageMarkerBegin+languageMarker.length+1)
+        return {language: language, 
+          source: codeCell.substring(languageMarkerBegin+languageMarker.length,languageMarkerEnd)}
+      }
+    }
+
+    function getCellLanguage(codeCell: string) {
+      let listOfLanguages = ["javascript", "python", "r"] 
+      for (var l = 0; l < listOfLanguages.length; l++) {
+        let languageMarker = "```".concat(listOfLanguages[l])
```

Comment:
I think we could do this more generally using another regular expression. If we match `codeCell` against a regular expression:

    ```[a-z]+

Then the result should be just the language, prefixed with three backticks - so if we drop the three backticks, we get a language, regardless of what it is...